micrometer-commons/src/main/java/io/micrometer/common/annotation/AnnotationUtils.java

```diff
@@ -36,18 +40,16 @@ private AnnotationUtils() {
 
     static List<AnnotatedParameter> findAnnotatedParameters(Class<? extends Annotation> annotationClazz, Method method,
             Object[] args) {
-        Annotation[][] parameters = method.getParameterAnnotations();
-        List<AnnotatedParameter> result = new ArrayList<>();
-        int i = 0;
-        for (Annotation[] parameter : parameters) {
-            for (Annotation parameter2 : parameter) {
-                if (annotationClazz.isAssignableFrom(parameter2.annotationType())) {
-                    result.add(new AnnotatedParameter(i, parameter2, args[i]));
-                }
+        Parameter[] parameters = method.getParameters();
```

Comment:
I appreciate the refactoring effort especially since the previous version bothers me a little:
- `Annotation[][] parameters` should have been `parameterAnnotations`
- `Annotation[] parameter` should have been `annotations`
- `Annotation parameter2` should have been `annotation`
- incrementing the index separately

This refactor fixes these concerns but it also introduces a few more.
I don't think that doing this is a good usage of the Stream API (it might be also less performant), what do you think about using a for loop with an index instead of an `IntStream`?
Something like this I guess:
```java
Parameter[] parameters = method.getParameters();
for (int i = 0; i < parameters.length; i++) {
    Parameter parameter = parameters[i];
    Annotation[] annotations = parameter.getAnnotationsByType(annotationClazz);
    if (annotations.length == 0) {
        return Collections.emptyList();
    }
    else {
        return Arrays.stream(annotations)
            .map(annotation -> new AnnotatedParameter(i, annotation, args[i]))
            .collect(Collectors.toList());
    }
}
```

Reply:
well, honestly i would argue about the approach :) Stream API-based approach could look like an overhead however it is always a matter of doing performance analysis first and then doing the conclusions especially now when we have corresponding tools like JMH. And unfortunately here we don't use extra libraries which would hide the additional complexity like StreamEx(`EntryStream.of(list)` will give you a stream of tuples where list element is associated with its index, or Guava's zipWithIndex). So this approach with IntStream doesn't seem to be breaking the main "rules" if such exist with regrads to how and where to use Stream API and in addition to that i remember i saw benchmark results where it was stated that on modern JVMs performance of Stream API is really close to `for` loops.

Anyway i reworked this place a bit to the suggested approach. Please see the result. Thank you so much for your comments and feedback